Fix NPE when importing SAML EntityDescriptor without SPSSODescriptor

PavlNekrasov opened the pull request:
resolves #44430

PavlNekrasov committed:
Fix NullPointerException when SPSSODescriptor is missing in SAML metadata

Signed-off-by: PavlNekrasov <95914807+PavlNekrasov@users.noreply.github.com>

mposolda commented:
@rmartinc Do you please have a chance to review this PR?

rmartinc commented on services/src/main/java/org/keycloak/protocol/saml/EntityDescriptorDescriptionConverter.java:
```diff
@@ -187,6 +187,7 @@ private static ClientRepresentation loadEntityDescriptors(InputStream is) {
         attributes.put(SamlConfigAttributes.SAML_SIGNATURE_ALGORITHM, SignatureAlgorithm.RSA_SHA256.toString());
         attributes.put(SamlConfigAttributes.SAML_AUTHNSTATEMENT, SamlProtocol.ATTRIBUTE_TRUE_VALUE);
         SPSSODescriptorType spDescriptorType = getSPDescriptor(entity);
+        if (spDescriptorType == null) return app;
```
This method is to create a SAML client via the  `SPSSODescriptor`. So, if the descriptor is not for an SP, maybe it's better to throw an exception. Because it makes no sense returning something that will be completely empty or filled with default values. Similar to the case when there is no entity descriptor a few lines before. Something like:

```java
if (spDescriptorType == null) {
    throw new RuntimeException("No SPSSODescriptorType defined in the entity descriptor file");
}
```

If you prefer you can return a `BadRequestException`. Because, just curiosity, what are you trying to import here? A IDPSSODescriptor or another thing?

rmartinc reviewed (changes requested):
I think that throwing an exception would be better than returning an empty client. WDYT?

PavlNekrasov committed:
Fix NullPointerException when SPSSODescriptor is missing in SAML metadata

throw `BadRequestException` instead of  `RuntimeException` in `loadEntityDescriptors` and  also if `spDescriptorType == null`

Signed-off-by: PavlNekrasov <95914807+PavlNekrasov@users.noreply.github.com>

rmartinc reviewed (approved):
Thanks @PavlNekrasov!

mposolda reviewed (approved):
@PavlNekrasov @rmartinc Thanks for the PR and review!
